routes/index.js

```diff
@@ -4,15 +4,30 @@ var Product = require('../models/product');
 
 /* GET home page. */
 router.get('/', function(req, res, next) {
-  res.render('index', { products: /* ??? */ });
+  res.render('index', { products: Product.all });
 });
 
 router.post('/cart', function(req, res, next) {
+  var temp = parseInt(req.body.id);
+  var product = Product.find(temp);
+  req.session.cart.push(product);
+  res.redirect('/');
+  // console.log(product);
   // How do we add an item to the cart?
 });
 
 router.delete('/cart', function(req, res, next) {
   // How can we remove an item from the cart?
+  var temp = parseInt(req.body.id);
+  // cart.delete(temp);
```

Comment:
Now that you've got the id from the body, how can you use this to delete it from the array? I'd look at `splice` or `filter`.